components/lib/calendar/Calendar.js

```diff
@@ -3070,29 +3070,29 @@ export const Calendar = React.memo(
                 isTypingRef.current = false;
 
                 // #3516 view date not updated when value set programatically
-                if (!visible && newDate) {
```

Comment:
You remove the !visible check?

also did you retest ticket #3516 ?

Reply:
I deleted `!visible` condition and changed to early return :)
I checked the #3516 ticket, It works as shown in the video below



https://github.com/primefaces/primereact/assets/37934668/4178371a-3f4e-4b20-ac39-b59c5fd723b3